api/v2/submitblock_json.go

```diff
@@ -0,0 +1,99 @@
+package v2
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	v1 "github.com/attestantio/go-builder-client/api/v1"
+	"github.com/attestantio/go-eth2-client/spec/bellatrix"
+	"github.com/attestantio/go-eth2-client/spec/capella"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/pkg/errors"
+)
+
+// submitBlockRequestJSON is the spec representation of the struct.
+type submitBlockRequestJSON struct {
+	Message                *v1.BidTrace                    `json:"message"`
```

Comment:
looks like some fields from the `ExecutionPayloadHeader` are duplicated in `BidTrace`

if we really want to squeeze performance out of this, shouldn't we remove the dupes?

Reply:
The reason I didn't  really want to do this is that the builder signature is over the complete bidtrace field, in order to allow exposing it to the public. It seems a bit messy to reconstruct the object that was signed over from a partial bidtrace and the header 🤔 

If there's strong desire to squeeze out every bit of performance, i'm open to discuss this and to change my mind. @michaelneuder wdyt?

Reply:
great points! i agree that since the builder signs over the bid trace, it is much simpler from the relay code perspective to have that full struct in the message for quick signature validation. also it has the added benefit of maintaining the bid trace logic throughout the code, rather than having to reconstruct the bid trace in the relay handler. In terms of repeated bytes we have five duplicate fields 

```
ParentHash     32 bytes
BlockHash      32 bytes
FeeRecipient   20 bytes
GasLimit       8 bytes
GasUsed        8 bytes
```

for a clean 100 bytes of duplicate data. Since we are still ~500 bytes below the MTU of 1500, IMO these 100 bytes are no problem.